### ISSUE TITLE
Daily 3x1 Widget "Oreo" and "Oreo (Google Sans)" view styles using wrong font

### ISSUE THREAD

**Issue by codenyte** [breezy-weather/breezy-weather#645]
### Steps to reproduce

1. Create a 3x1 Daily widget
2. Select the "Oreo (Google Sans)" view style

### Expected behavior

The "Oreo (Google Sans)" view style should use the Google Sans font, and the "Oreo" view style should use the system default font.

### Actual behavior

When selecting "Oreo (Google Sans)", the correct font is shown in the preview while configuring the widget, but the widget itself on the home screen uses the system default font instead of Google Sans.  But when I select the "Oreo" view style, the widget does actually use Google Sans, and it shows the default font in the preview.

### Weather source used

Open-Meteo

### Breezy Weather version

4.6.14-beta

### Android version

Android 14

### Device

Google Pixel 6a

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I read the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document and followed instructions if relevant.
- [X] I have updated the app to **[version 4.6.14 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by min7-i** [comment]
Apparently, the option to set a custom font for widgets was removed with an Android update in June 2021 (see [issue](https://issuetracker.google.com/issues/190444596#comment7)). The most important part:

> This is an intended change. Allowing custom fonts in app widgets was an unintended behavior and could also expose a security bug. In june update we have fixed that flaw.

I tested this on my two devices:
| Style | Location | Pixel 6a (Android 14) | LG G6 (Android 9) |
| --- | --- | --- | --- |
| Oreo | preview | system font | system font |
| Oreo | home screen | system font | system font |
| Oreo (Google Sans) | preview  | Google Sans | Google Sans |
| **Oreo (Google Sans)** | **home screen** | **system font** | **Google Sans** |

> But when I select the "Oreo" view style, the widget does actually use Google Sans, and it shows the default font in the preview.

@codenyte This would be a real curiosity. The font Google Sans is correctly defined for the Oreo (Google Sans) layouts only ([1](https://github.com/breezy-weather/breezy-weather/blob/main/app/src/main/res/layout/widget_day_oreo_google_sans.xml#L23), [2](app/src/main/res/layout/widget_day_oreo_google_sans.xml)) and this mix-up didn't happen on my devices. Additionally, as you are using Android 14 it should use the system font for the widget on your home screen anyway (see linked issue above). Hence, I'd appreciate it if you could retest it and post a screenshot/screen recording if you're able to recreate it.

**Comment by papjul** [comment]
> Does Breezy Weather come with Google Sans[…]?

It's bundled in the app.

Thank you for your analysis, @min7-i!

If support was removed 3 years ago (except for custom launchers which might not have the security fix?), I think the good thing to do is to remove "Oreo (Google Sans)" option and only keep "Oreo", or we will only keep getting more and more reports that it doesn't work and there's nothing we can do. This will also save 1,1 Mo of resources as we can remove all font files which are not used anywhere else ;)

**Comment by min7-i** [comment]
> If support was removed 3 years ago (except for custom launchers which might not have the security fix?), I think the good thing to do is to remove "Oreo (Google Sans)" option and only keep "Oreo"

Agreed. It's the only view style in all of the 13 widgets that uses a custom font and there was already a filter that hid the Google Sans option for devices running Android < version 8:
https://github.com/breezy-weather/breezy-weather/blob/5456eaee4619a822f3701610ca661d564e2f4aeb/app/src/main/java/org/breezyweather/remoteviews/config/DayWidgetConfigActivity.kt#L31

**Comment by min7-i** [comment]
@codenyte Thanks for the update and mentioning the launchers you used. That helped a lot. No need to make a screen recording anymore.

I had to fiddle around a little bit to download Ruthless launcher; probably because it was last updated in 2022 and the target API level is too low for Android 14. Anyway, after successfully installing it on my Pixel 6a, I could confirm your test results.

I checked the settings of Ruthless Launcher and it allows you to set a custom font for your home screen with the default being Google Sans. From what I could tell the launcher tries to apply this font to all widgets that display text.

However, it appears that if a widget has its own font defined, the launcher will not apply the launcher font. So on older Android versions you see the widget's custom font (tested on my LG G6). On newer Android versions it will fall back to the system font. And as a result you have a confusing mix of fonts :confused:
